lib/model/QubitJob.php

```diff
@@ -275,7 +275,7 @@ public static function runJob($jobName, $jobParams = array())
       // so the resources modified are persisted before the assigned worker starts
       // processing the task. If we don't do this now the transaction will be committed
       // once this request is processed but not before the worker hits the database.
-      $connection = QubitTransactionFilter::getConnection();
+      $connection = Propel::getConnection();
```

Comment:
I think both return the same connection, but in this case using `QubitTransaction` makes clearer from where the transaction comes. The comment may be good enough, but I'd leave this one like it was, just in case.

Reply:
When I saw `QubitTransactionFilter::getConnection();` my assumption was that transaction is started by the `getConnection()` method call, whereas `Propel::getConnection();` doesn't start a transaction, which is false.  I think having a separate `QubitTransactionFilter::getConnection();` method that is basically just a wrapper for `Propel::getConnection();` is misleading and unnecessary.  I'm tempted to get rid of `QubitTransactionFilter::getConnection();` altogether, but I think it does help legibility and factoring in `QubitTransactionFilter::getConnnection();`.  Maybe it's best to make `QubitTransactionFilter::getConnection();` a protected method to make it clear that it's not part of the public signature.

Reply:
The transaction is started after getting the connection, when `$conn->beginTransaction();` is called. In the end `QubitTransactionFilter::getConnection()` calls `Propel::getConnection()` if it doesn't have a connection already. In both cases (I am almost sure) the same connection is obtained so I think it's not a big deal to use one or the other but, to be sure, calling `QubitTransactionFilter::getConnection()` gives me a bit more confident about having the same connection that started the transaction.